### PR TITLE
Show error when some action fails in banned competitor page

### PR DESCRIPTION
**Pull request by danieljames-dj**
This PR also edits the Errored component to add the error message.

### CODE REVIEW

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx [hunk 1]
```diff
@@ -27,7 +29,9 @@ export default function BanendCompetitorForm({
     }, () => {
       sync();
       closeForm();
-    }, { method: 'POST' });
+    }, { method: 'POST' }, (error) => {
+      setFormError(error);
+    });
```
Comment (gregorbg) [inline app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx:34]: Just pass in `setFormError` as a callback reference?

Comment (gregorbg) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx:34]:
```suggestion
    }, { method: 'POST' }, setFormError);
```

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx [hunk 2]
```diff
@@ -36,12 +40,15 @@ export default function BanendCompetitorForm({
     }, () => {
       sync();
       closeForm();
-    }, { method: 'PATCH' });
+    }, { method: 'PATCH' }, (error) => {
+      setFormError(error);
+    });
```
Comment (gregorbg) [inline app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx:45]: see my other comment

File: app/webpacker/components/Requests/Errored.js [hunk 3]
```diff
@@ -3,21 +3,25 @@ import React from 'react';
...
+            Something went wrong while loading the data
+            {componentName && (
+            <>
+              {' '}
+              for the component &apos;
+              {componentName}
+              &apos;
+            </>
+            )}
+          </>
```
Comment (gregorbg) [inline app/webpacker/components/Requests/Errored.js:24]: I don't think the error message alone is always clear enough. Remember that this is a pretty generic component that's being used all over the place, and error messages are not always self-contained.
So perhaps stick with the existing text (pre your change) and _append_ the error text additionally, if present?

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx (hunk 1 above)
Comment (danieljames-dj) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx]: Done

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx (hunk 2 above)
Comment (danieljames-dj) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitorForm.jsx]: Done

**Review by gregorbg — APPROVED** [review]

**Pull request merged**